File: Data Structures & Algorithms/car-fleet/submission-1.cpp
class Solution {
public:
    int carFleet(int target, vector<int>& position, vector<int>& speed) {
        
        vector<pair<int,float>> imp;

        for(int i=0;i<position.size();i++){
            float time = (float)(target-position[i])/speed[i];
            cout<<time<<endl;
            imp.push_back(make_pair(position[i],time));
        }
        sort(imp.begin(),imp.end());

        stack<float> fleets;
        int ans = 0;
        for(int i = position.size()-1;i>=0;i--){

            cout<<imp[i].second<<endl;
            if(fleets.empty()){
                fleets.push(imp[i].second);
            }
            else if(fleets.top()<imp[i].second){
                fleets.pop();
                ans++;
                fleets.push(imp[i].second);
            }
        }

        ans++;
        return ans;


    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        
        def req(nums,target,i,curr):
            if target==0:
                ans.append(curr.copy())
                return
            if target<0 or i==len(nums):
                return
            req(nums,target,i+1,curr)
            curr.append(nums[i])
            req(nums,target-nums[i],i,curr)
            curr.pop()
        ans = []
        curr = []
        i = 0
        req(nums,target,i,curr)
        return ans